src/services/redisSessionStorage.ts

```diff
@@ -2,26 +2,37 @@
  * This service uses the Redis client to store and retrieve session information.
  */
 
-import { Either, left } from "fp-ts/lib/Either";
+import { Either, left, right } from "fp-ts/lib/Either";
 import * as redis from "redis";
 import { User } from "../types/user";
 import { extractUserFromJson } from "../types/user";
 import { ISessionStorage } from "./iSessionStorage";
+import TokenService from "./tokenService";
+
+const SESSION_NOT_FOUND_MESSAGE = "Session not found";
+const REDIS_DATA_FIELD = "data";
+const REDIS_TIMESTAMP_FIELD = "data";
 
 export default class RedisSessionStorage implements ISessionStorage {
   constructor(
     private readonly redisClient: redis.RedisClient,
-    private readonly tokenDuration: number
+    private readonly tokenDuration: number,
+    private readonly tokenService: TokenService
   ) {}
 
   /**
    * {@inheritDoc}
    */
   public set(token: string, user: User): void {
-    // Set key to hold the string value. This data is set to expire (EX) after
-    // `this.tokenDuration` seconds.
-    // @see https://redis.io/commands/set
-    this.redisClient.set(token, JSON.stringify(user), "EX", this.tokenDuration);
+    // Set a key to hold the fields value.
+    // @see https://redis.io/commands/hmset
+    this.redisClient.hmset(
+      token,
+      REDIS_DATA_FIELD,
```

Comment:
maybe we can use hmset(hash, object) to pass a typed object ?

Reply:
usually is better to store an hash instead of a serialized object in redis

Reply:
I agree but according to the client docs (if it's this one) you can pass an object and
"The properties and values of this Object will be set as keys and values in the Redis hash.":
http://redis.js.org/#api-friendlier-hash-commands-clienthmsethash-obj-callback
(isn't it the same behavior as the current one ?)